src/test/java/org/jitsi/meet/test/base/FailureListener.java

```diff
@@ -0,0 +1,362 @@
+    {
+        saveMeetDebugLog(testInstance.participant1.getDriver(),
+            fileNamePrefix + "-meetlog-owner.json");
+
+        if(testInstance.participant2 != null)
+        {
+            saveMeetDebugLog(testInstance.participant2.getDriver(),
+                fileNamePrefix + "-meetlog-participant.json");
+        }
+
+        if(testInstance.participant3 != null)
+        {
+            saveMeetDebugLog(testInstance.participant3.getDriver(),
+                fileNamePrefix + "-meetlog-third.json");
+        }
+    }
+
+    /**
+     * Saves the log from meet. Normally when clicked it is saved in Downloads
+     * if we do not find it we skip it.
+     */
+    private void saveMeetDebugLog(WebDriver driver, String fileName)
+    {
+        try
+        {
+            Object log = ((JavascriptExecutor) driver)
+                .executeScript(
+                    "try{ "
+                        + "return JSON.stringify(APP.conference.getLogs(), null, '    ');"
+                        + "}catch (e) {}");
+
+            if(log == null)
+                return;
+
+            FileUtils.write(
+                new File(outputLogsParentFolder, fileName),
+                (String)log);
+        }
+        catch (Exception e)
+        {
+            //e.printStackTrace();
+        }
+    }
+
+    /**
+     * Saves browser console logs.
+     */
+    private void saveBrowserLogs(
+        String fileNamePrefix, AbstractBaseTest testInstance)
+    {
+        saveBrowserLogs(
+            testInstance.participant1.getDriver(),
+            fileNamePrefix, "-console-owner", ".log",
+            testInstance.participant1.getType());
+
+        if(testInstance.participant2 != null)
+            saveBrowserLogs(
+                testInstance.participant2.getDriver(),
+                fileNamePrefix, "-console-secondParticipant", ".log",
+                testInstance.participant2.getType());
+
+        if(testInstance.participant3 != null)
+            saveBrowserLogs(
+                testInstance.participant3.getDriver(),
+                fileNamePrefix, "-console-thirdParticipant", ".log",
+                testInstance.participant3.getType());
+    }
+
+    /**
+     * Saves current java thread dump.
+     */
+    private void saveThreadDump(String fileNamePrefix)
+    {
+        StringBuilder dump = new StringBuilder();
+        ThreadMXBean tbean = ManagementFactory.getThreadMXBean();
+        ThreadInfo[] threads
+            = tbean.getThreadInfo(tbean.getAllThreadIds(), 150);
+        for (ThreadInfo tinfo : threads)
+        {
+            dump.append(tinfo);
+            dump.append("\n");
+        }
+
+        try
+        {
+            BufferedWriter out = new BufferedWriter(new FileWriter(
+                new File(outputLogsParentFolder,
+                    fileNamePrefix + ".tdump")));
+            out.write(dump.toString());
+            out.flush();
+            out.close();
+        }
+        catch (IOException e){}
```

Comment:
ignored exception, you can also use new java 8 syntax in all file operations:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html